### DISCUSSION TITLE
解决 MacBook 键盘双击问题 | SF-Zhou's Blog

### DISCUSSION THREAD

**Discussion by SF-Zhou** [SF-Zhou/sf-zhou.github.io#165] [Announcements]
https://sf-zhou.github.io/productivity/solve_macbook_typing_double.html

**Comment by SF-Zhou** [comment]
上周拿到 Apple Store 送修了，服务很棒。今天取回了机器，重新换了整个键盘以及 Touch Bar，和电池。苹果也是很给力了👍。所以，遇到上面的问题，直接送修好了。

**Comment by AbnerVictor** [comment]
我也遇到了和你一样的问题，我的问题出在“x”键上，上周也送修了。且不说结果如何，你解决问题的思路很有趣，点赞

**Comment by xiaolanglanglang** [comment]
感谢您的代码…………
救我脱离出苦海………… Apple Store 太难约了……而且实在是没办法脱离这个笔记本太久…………

**Comment by SF-Zhou** [comment]
@gy911201 可以先将就着🤓然后代码里 ignore 的时限也可以调一调。

**Comment by zibo-ye** [comment]
拿到新Macbook之后发现的第一个问题就是space双击，用作者的代码，把时限调到100ms，基本完美解决了这个问题(主要还是懒得去Apple Store)

**Comment by xiaolanglanglang** [comment]
终于抽到一个比较空的时间把键盘送修了……主要是现在除了双击还开始失去响应了…………………………
然后换个键盘给我说要换 7 天………………醉了………………

**Comment by Jaydentlee** [comment]
感谢博主提供的解决方案！！我去年八月在美国买的2017 MacBook pro，今年7月开始出现B键双击，最近连N键也开始渐渐出现问题。。。但目前在国内，也过了保修期，送修是一个大问题。博主的方案拯救了我！！

**Comment by xiaolanglanglang** [comment]
> @Jaydentlee
> 感谢博主提供的解决方案！！我去年八月在美国买的2017 MacBook pro，今年7月开始出现B键双击，最近连N键也开始渐渐出现问题。。。但目前在国内，也过了保修期，送修是一个大问题。博主的方案拯救了我！！

苹果前段时间发布了召回计划，你可以关注一下，看看能不能在中国召回维修

**Comment by SF-Zhou** [comment]
> > @Jaydentlee
> > 感谢博主提供的解决方案！！我去年八月在美国买的2017 MacBook pro，今年7月开始出现B键双击，最近连N键也开始渐渐出现问题。。。但目前在国内，也过了保修期，送修是一个大问题。博主的方案拯救了我！！
> 
> 苹果前段时间发布了召回计划，你可以关注一下，看看能不能在中国召回维修

对的，四年之内都可以，https://www.apple.com/cn/support/keyboard-service-program-for-macbook-and-macbook-pro

**Comment by Jaydentlee** [comment]
> @SF-Zhou
> > > @Jaydentlee
> > > 感谢博主提供的解决方案！！我去年八月在美国买的2017 MacBook pro，今年7月开始出现B键双击，最近连N键也开始渐渐出现问题。。。但目前在国内，也过了保修期，送修是一个大问题。博主的方案拯救了我！！
> > 
> > 苹果前段时间发布了召回计划，你可以关注一下，看看能不能在中国召回维修
> 
> 对的，四年之内都可以，https://www.apple.com/cn/support/keyboard-service-program-for-macbook-and-macbook-pro

感谢！已联系苹果，说是全球联保，准备周末送过去看看。不过一送修就要好几天，最近事情又多，只能把东西迁移到台式PC凑合一段时间了。。。

**Comment by Cindyshen725** [comment]
@SF-Zhou
你好博主，我是2018款Mac Air，刚买三个多月，最近出现space双击，也可以用这段代码嘛？要怎么改啊。谢谢啦

**Comment by SF-Zhou** [comment]
> @Cindyshen725
> @SF-Zhou
> 你好博主，我是2018款Mac Air，刚买三个多月，最近出现space双击，也可以用这段代码嘛？要怎么改啊。谢谢啦

space 的 keycode 是 49，所以把代码中 `if (keycode == 11 /* b */ || keycode == 45 /* n */) {` 换成 `if (keycode == 49) {` 即可。建议直接送修，自己折腾还是有点麻烦的。

**Comment by Cindyshen725** [comment]
谢谢啊。国外送修人工效率巨低……刚买了三个月的机器就要被拆心有点疼 哈哈哈。

**Comment by chuanqizhang** [comment]
您好，请问怎么具体用nohup实现后台执行呢？（关闭所有终端窗口仍能保持后台执行）谢谢！

**Comment by chuanqizhang** [comment]
> 您好，请问怎么具体用nohup实现后台执行呢？（关闭所有终端窗口仍能保持后台执行）谢谢！

update一下：貌似nohup ./processname &就可以了，如有问题望请指正。

**Comment by SF-Zhou** [comment]
> @chuanqizhang
> 您好，请问怎么具体用nohup实现后台执行呢？（关闭所有终端窗口仍能保持后台执行）谢谢！

`nohup ./kill_double_typing &`，很容易就能搜到的

**Comment by chuanqizhang** [comment]
> > @chuanqizhang
> > 您好，请问怎么具体用nohup实现后台执行呢？（关闭所有终端窗口仍能保持后台执行）谢谢！
> 
> `nohup ./kill_double_typing &`，很容易就能搜到的

感谢感谢。这段代码really works well

**Comment by cbxgithub** [comment]
您好，我直接用g++编译报错：Undefined symbols for architecture x86_64，请问你是用什么方式编译的？

**Comment by SF-Zhou** [comment]
> @cbxgithub
> 您好，我直接用g++编译报错：Undefined symbols for architecture x86_64，请问你是用什么方式编译的？

代码头部写了编译命令

**Comment by cbxgithub** [comment]
> @SF-Zhou
> > @cbxgithub
> > 您好，我直接用g++编译报错：Undefined symbols for architecture x86_64，请问你是用什么方式编译的？
> 
> 代码头部写了编译命令

多谢！

**Comment by Axure** [comment]
太厉害了啵
